rust/src/array.rs

```diff
@@ -239,6 +241,67 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
     }
 }
 
+macro_rules! def_numeric_math_ops {
```

Comment:
Instead of macro, you can do:
```
impl<T: ArrowNumericType> PrimitiveArray<T>
where
    T::Native: Add<Output = T::Native>
        + Sub<Output = T::Native>
        + Mul<Output = T::Native>
        + Div<Output = T::Native>
        + Zero,
{
  // actual impl
}
```